### ISSUE TITLE
Add easy way to modify links and add toggle options.

### ISSUE THREAD

**Issue by phpbits** [WordPress/gutenberg#17556]
**Is your feature request related to a problem? Please describe.**
Related to : https://github.com/WordPress/gutenberg/pull/16609

**Describe the solution you'd like**
It would be really great if we can easily extend the link format. Adding "rel" attributes value requires us to create a new format and deregister the core one. There's a need for option to add toggles on the link options as well.

**Describe alternatives you've considered**
Right now I've created a new link format on [EditorsKit](https://wordpress.org/plugins/block-options/). But I don't want this solution of mine to cause any issues in the future if in any case there will be API changes. Thanks!

![Screen Capture on 2019-09-16 at 12-42-40](https://user-images.githubusercontent.com/3365507/65531671-fdbba900-df2c-11e9-81ca-66e25d9e9160.gif)

**Comment by carlenzoliyuri** [comment]
IMHO

"No follow" label is not really simple for all users. "Paid link" is more simple.

And rel for this is ideal rel="nofollow sponsored" for compatibility to all search engine, now. When sponsored be a W3 standard nofollow is useless.

Only one option is good for me.

**Comment by bemyhre** [comment]
would be awesome.

**Comment by blogtutor** [comment]
Yes, please! There is a desperate need for this. Having this ability will increase the Block Editor adoption significantly.

We have hundreds of clients who have been waiting to switch to the Block Editor, and literally **the only thing preventing our clients from using the block editor is the inability to add "nofollow" tags to links.**  (Most of our clients are not comfortable editing HTML.)

I am thrilled that you've been able to add this to EditorsKit -- but I agree, it would be much better to make feature part of Core, and/or to make it so that a smaller plugin can be added (similar to plugins like "Ultimate Nofollow," which only works on Classic Editor), to add this feature.

**Comment by keepact** [comment]
I was able to add this option by writing a lot of code, it didn't have to be this way. I followed the concepts of this PR [#16609](https://github.com/WordPress/gutenberg/pull/16609) but I'm waiting for new ways to extend this format!

**Comment by tw2113** [comment]
Maybe instead of a basic toggle that is hardcoded to nofollow, and sponsored, perhaps a sidebar text field to provide rel attributes to be applied. For example, there's still the very useful `rel="me"` option to indicate relation of linked urls.

Thus far at least, i can get to the link markup with the "edit as html" option, but that's nowhere near as intuitive for others who may not think of that. I also have to be careful not to touch the link again because I assume it'd get lost. A dedicated setting somehow would be nice.

**Comment by paaljoachim** [comment]
I went ahead and added a new issue: 
https://github.com/WordPress/gutenberg/issues/23011

**Comment by Mamaduka** [comment]
This is a duplicate of #11599.
